modules/speicher_e3dc/e3dc.py

```diff
@@ -1,56 +1,71 @@
-#!/usr/bin/python
-import sys
-# import os
-# import time
-# import getopt
-# import socket
-# import ConfigParser
-import struct
-# import binascii
-from pymodbus.client.sync import ModbusTcpClient
+#!/usr/bin/python3
+import logging
+from statistics import mean
+from typing import Iterable
 
-ipaddress = str(sys.argv[1])
-addext = int(sys.argv[2])
+from pymodbus.constants import Endian
 
-client = ModbusTcpClient(ipaddress, port=502)
+from helpermodules.cli import run_using_positional_cli_args
+from helpermodules.log import setup_logging_stdout
+from modules.common.component_state import InverterState, BatState
+from modules.common.modbus import ModbusClient, ModbusDataType
+from modules.common.store import get_inverter_value_store, get_bat_value_store
+from modules.common.simcount import SimCountFactory
+from modules.common.store.ramdisk import files
 
-# battsoc
-resp= client.read_holding_registers(40082,1,unit=1)
-value1 = resp.registers[0]
-all = format(value1, '04x')
-final = int(struct.unpack('>h', all.decode('hex'))[0]) 
-f = open('/var/www/html/openWB/ramdisk/speichersoc', 'w')
-f.write(str(final))
-f.close()
-# print "hausverbrauch"
-# resp= client.read_holding_registers(40071,2,unit=1)
-# value1 = resp.registers[0]
-# value2 = resp.registers[1]
-# all = format(value2, '04x') + format(value1, '04x')
-# final = int(struct.unpack('>i', all.decode('hex'))[0])
-# print final
-# pv punkt
-ext = 0
-if addext == 1:
-    resp= client.read_holding_registers(40075,2,unit=1)
-    value1 = resp.registers[0]
-    value2 = resp.registers[1]
-    all = format(value2, '04x') + format(value1, '04x')
-    ext = int(struct.unpack('>i', all.decode('hex'))[0])
-resp= client.read_holding_registers(40067,2,unit=1)
-value1 = resp.registers[0]
-value2 = resp.registers[1]
-all = format(value2, '04x') + format(value1, '04x')
-final = int(struct.unpack('>i', all.decode('hex'))[0]) * -1
-f = open('/var/www/html/openWB/ramdisk/pvwatt', 'w')
-f.write(str(final+ext))
-f.close()
-#battleistung
-resp= client.read_holding_registers(40069,2,unit=1)
-value1 = resp.registers[0]
-value2 = resp.registers[1]
-all = format(value2, '04x') + format(value1, '04x')
-final = int(struct.unpack('>i', all.decode('hex'))[0])
-f = open('/var/www/html/openWB/ramdisk/speicherleistung', 'w')
-f.write(str(final))
-f.close()
+log = logging.getLogger("E3DC Battery")
+
+def update_e3dc_battery(addresses: Iterable[str],read_external: int,pv_other: int):
```

Comment:
Jetzt wo du in Zeile 65 `pv_other = pvmodul != "none"` hast müsste der Typ hier noch ein `bool` werden:
```suggestion
def update_e3dc_battery(addresses: Iterable[str], read_external: int, pv_other: bool):
```

Reply:
gemacht